### PR TITLE
TG-632 - Route Auth Notification via the Distributor

### PR DESCRIPTION
**Pull request by theunissens**

### CODE REVIEW

File: imqsauth/http.go [hunk 1]
```diff
@@ -1115,72 +1105,26 @@ func getUserIdOrIdentity(r *httpRequest) (string, authaus.UserId, error) {
...
-	}
-
-	if closeConnection {
-		ws.Close()
-		return
+func (central *ImqsCentral) broadcastToAllSubscribers(msg string) {
+	notificationReqJson := &notificationRequestJson{}
+	notificationReqJson.Channel = "authNotifications"
+	notificationReqJson.Msg = msg
+	jsonString, err := json.Marshal(notificationReqJson)
```
Comment (reinhartfimqs) [inline imqsauth/http.go]: You can change the json.Marshal(notificationReqJson) into json.Marshal(&notificationRequestJson{"authNotifications", msg}) if you want to make it simpler

File: imqsauth/http.go [hunk 2]
```diff
@@ -1115,72 +1105,26 @@ func getUserIdOrIdentity(r *httpRequest) (string, authaus.UserId, error) {
...
-
-	central.subscriberLock.RUnlock()
-
-	// now delete the recorded id's
-	for _, id := range deleteIDs {
-		closeAndDeleteWebSocket(central, id)
+	defer resp.Body.Close()
+	respBody, _ := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode != http.StatusOK {
+		central.Central.Log.Errorf("Error notifying distributor about auth group changes: %v\n%v", err, string(respBody))
```
Comment (reinhartfimqs) [inline imqsauth/http.go]: This falls inline with the other story I just did where devops requested we change as many Errorf statements to Warnf statements so that loggly does not report on everything, can we change these?

Comment (theunissens) [inline imqsauth/http.go]: Yes - my bad, I was in the habbit. I will change them.

File: imqsauth/http.go [hunk 3]
```diff
@@ -1115,72 +1105,26 @@ func getUserIdOrIdentity(r *httpRequest) (string, authaus.UserId, error) {
...
-
-	for id, ws := range central.roleChangeSubscribers {
-		err := websocket.Message.Send(ws, msg)
-		if err != nil {
-			// only record the id, since we already have a read lock active at this point
-			deleteIDs = append(deleteIDs, id)
-		}
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		central.Central.Log.Errorf("Error notifying distributor about auth group changes: %v", err)
```
Comment (reinhartfimqs) [inline imqsauth/http.go]: Is it a good idea to have all of the error messages reporting the same message with a different error code? I am thinking that it might be a little confusing trying to figure out on which err it is failing

File: imqsauth/http.go [hunk 4]
```diff
@@ -933,11 +923,11 @@ func httpHandlerSetUserGroups(central *ImqsCentral, w http.ResponseWriter, r *ht
...
 	}
 	*/
-	central.broadcastToAllSubscribers(createNotificationMessage(notificationChannel, notificationType, strconv.FormatInt(int64(userId), 10)))
+	central.broadcastToAllSubscribers(createNotificationMessage(strconv.FormatInt(int64(userId), 10)))
 }
 
-func createNotificationMessage(channelName string, typeName string, messageContent string) string {
-	return fmt.Sprintf("%v:%v:%v", channelName, typeName, messageContent)
+func createNotificationMessage(messageContent string) string {
+	return fmt.Sprintf("%v:%v", "permissions_changed", messageContent)
```
Comment (reinhartfimqs) [inline imqsauth/http.go]: Why did you remove the typeName? I am just thinking that if we want to send a message with another notification type we either have to change this one or create a new one

**Comment by reinhartfimqs** [comment]
LGTM
